rt/frontend/jaxrs/src/test/java/org/apache/cxf/jaxrs/provider/ProviderFactoryTest.java

```diff
@@ -820,6 +822,40 @@ public void testSetSchemasFromClasspath() {
         assertNotNull("schema can not be read from classpath", s);
     }
 
+    @Test
+    public void testSortByPriority() {
+        ProviderFactory pf = ServerProviderFactory.getInstance();
+        List<Object> providers = new ArrayList<>();
+        BookReaderWriter bookHandler = new BookReaderWriter();
+        providers.add(bookHandler);
+        HighPriorityBookReaderWriter highPriorityBookHandler = new HighPriorityBookReaderWriter();
+        providers.add(highPriorityBookHandler);
+        pf.setUserProviders(providers);
+        assertSame(highPriorityBookHandler,
+                   pf.createMessageBodyReader(Book.class, Book.class, new Annotation[]{},
+                                              MediaType.APPLICATION_XML_TYPE, new MessageImpl()));
+        assertSame(highPriorityBookHandler,
+                   pf.createMessageBodyWriter(Book.class, Book.class, new Annotation[]{},
+                                              MediaType.APPLICATION_XML_TYPE, new MessageImpl()));
+        
+        // do the same thing again but add the providers in the reverse order to ensure add order
```

Comment:
Would you mind please to extract the second part of the test case to separate method, for example `public void testSortByPriorityReversed()`? Thanks.